### PR TITLE
Add blog templates

### PR DESCRIPTION
**Pull request by maya**
This adds the blog index, single blog post (large, small, and no feature image), and tag index page.

More features and functionality to come around the blog...

### CODE REVIEW

File: _includes/related-posts.html [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  {% for post in site.related_posts %}
+    {% assign postHasSimilar = false %}
+    {% for tag in post.tags %}
+      {% for thisTag in page.tags %}
+        {% if postHasSimilar == false and hasSimilar.size < 3 and post != page and tag == thisTag %}
+          {% if hasSimilar.size == 0 %}
+          <h2>Related Posts</h4>
+          <ul>
+          {% endif %}
+          <li class="usa-width-one-third">
```
Comment (elainekamlley) [inline _includes/related-posts.html]: `<li>` indent in one step

File: _includes/related-posts.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+              </p>
+              <span class="post-tags" itemprop="keywords">
+                {% for tag in post.tags %}
+                  <a class="usa-label" href="{{ site.baseurl }}/{{ site.tag_dir }}/{{ tag | slugify }}/">{{ tag }}
+                  </a>
+                {% endfor %}
+              </span>
+            </article>
+          </li>
+          {% capture hasSimilar %}{{ hasSimilar }}*{% endcapture %}
```
Comment (elainekamlley) [inline _includes/related-posts.html]: Interested to know more about this. Seems like `hasSimilar` is capturing itself?

File: _layouts/post.html [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+{% endif %}
+
+<div class="usa-grid">
+
+  <a class="back-blog link-arrow-left" href="{{ site.baseurl }}/blog/">Blog</a>
+  <article class="post" itemscope itemtype="http://schema.org/BlogPosting">
+    <header class="post-header">
+      <h1 class="post-title" itemprop="name headline">{{ page.title }}</h1>
+      <p class="post-byline">
+      {% if page.authors %}
```
Comment (elainekamlley) [inline _layouts/post.html]: Noticed that `authors` is in two places. Should we make it an `_include`?

File: _layouts/tag_index.html [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    <ul class="post-list">
+      {% for post in page.posts %}
+        <li>
+          <div class="usa-width-two-thirds">
+            <span class="post-date">{{ post.date | date: "%B %-d, %Y" }}</span>
+            <h2>
+              <a class="post-link" href="{{ post.url | prepend: site.baseurl }}">{{ post.title }}</a>
+            </h2>
+            <p class="post-byline">
+            {% if post.authors %}
```
Comment (elainekamlley) [inline _layouts/tag_index.html]: Same question as above about `authors`

File: _layouts/tag_page.html [hunk 5]
```diff
@@ -0,0 +1,43 @@
+---
```
Comment (elainekamlley) [inline _layouts/tag_page.html]: What is the difference between `tag_index` and `tag_page`?

File: _sass/_components/blog-index.scss [hunk 6]
```diff
@@ -0,0 +1,111 @@
+// Blog index
```
Comment (elainekamlley) [inline _sass/_components/blog-index.scss:1]: Could some of these break down to smaller components? Like RSS feed for example?

File: _sass/_components/blog-post.scss [hunk 7]
```diff
@@ -0,0 +1,197 @@
+// Blog post
```
Comment (elainekamlley) [inline _sass/_components/blog-post.scss:1]: Similar comment about `blog_index`. Could we break it into smaller components? Could help to reduce over nesting.

File: _sass/_components/blog-post.scss [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+      @include margin($section-margins null);
+    }
+
+    figure {
+      @include margin($section-margins null);
+      text-align: center; // center image if less than full width
+
+      img {
+        @include margin(0 null);
+      }
```
Comment (elainekamlley) [inline _sass/_components/blog-post.scss:37]: For example - this `img` tag is nested four deep.

File: _sass/_components/blog-post.scss [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+    }
+
+    h5 {
+      color: $color-base;
+      margin-top: 1rem;
+    }
+  }
+}
+
+.social_bar {
```
Comment (elainekamlley) [inline _sass/_components/blog-post.scss]: This could be its own component.

File: _sass/_components/layout.scss [hunk 10]
```diff
@@ -48,11 +48,51 @@
...
 
     &:hover {
       color: $color-bright-hover;
     }
   }
+  .usa-button,
+  .usa-button-primary,
+  .usa-button:visited,
+  .usa-button-primary:visited,
+  button,
```
Comment (elainekamlley) [inline _sass/_components/layout.scss:64]: Should this be in `buttons.scss`?

File: _sass/_components/post-feature.scss [hunk 11]
```diff
@@ -12,19 +16,14 @@
...
-      }
-
-      &:last-child {
-        @include link-arrow-right(medium/arrow-right);
-        font-size: $small-font-size;
-      }
     }
 
-    h3 {
+    .posts_feature-heading {
```
Comment (elainekamlley) [inline _sass/_components/post-feature.scss]: Does this need to be nested in `.post_feature`?

File: blog/index.html [hunk 12]
```diff
@@ -1,9 +1,97 @@
...
+
+    {% for post in paginator.posts %}
+      <li>
+        <div class="usa-width-two-thirds">
+          <span class="post-date">{{ post.date | date: "%B %-d, %Y" }}</span>
+          <h2>
+            <a class="post-link" href="{{ post.url | prepend: site.baseurl }}">{{ post.title }}</a>
+          </h2>
+          <p class="post-byline">
+          {% if post.authors %}
```
Comment (elainekamlley) [inline blog/index.html]: Author _include?

File: _layouts/tag_page.html (hunk 5 above)
Comment (maya) [inline _layouts/tag_page.html]: 1. **Tag index** is a page with all the matching tag results = https://federalist.18f.gov/preview/18F/beta.18f.gov/add-blog/tags/agency-work/
2. **Tag page** is a list of all the tags = https://federalist.18f.gov/preview/18F/beta.18f.gov/add-blog/tags/

Maybe those titles don't make sense. Maybe it should be tag results for `#1` and tag index for `#2`?

File: _sass/_components/blog-index.scss (hunk 6 above)
Comment (maya) [inline _sass/_components/blog-index.scss:1]: Yes, RSS link moved to buttons.scss in c134cab, post list and pagination made into their own SCSS files in 3770842.

File: _sass/_components/blog-post.scss (hunk 9 above)
Comment (maya) [inline _sass/_components/blog-post.scss]: Social bar component removed after discussing with design team.

File: _layouts/post.html (hunk 3 above)
Comment (maya) [inline _layouts/post.html]: Made it into an include in 1e47713. However, I don't know how to make it so it shows `{% if page.authors %}` and `{% if post.authors %}` for the different pages. So post.html uses `{% if page.authors %}`, while the tag page and blog index page uses `{% if post.authors %}`. Any idea how to do this?

File: _sass/_components/blog-post.scss (hunk 8 above)
Comment (maya) [inline _sass/_components/blog-post.scss:37]: It's nested in the page `.single-post` inside `article` - I think these are both necessary for this style.

File: _sass/_components/layout.scss (hunk 10 above)
Comment (maya) [inline _sass/_components/layout.scss:64]: It's a button style but for `.background-dark` layout type, so I'm not sure if it should be moved. It's nice to see all the changes for that layout together.

File: _sass/_components/post-feature.scss (hunk 11 above)
Comment (maya) [inline _sass/_components/post-feature.scss]: It does not. Fixed in 6b3dbb7.

File: blog/index.html (hunk 12 above)
Comment (maya) [inline blog/index.html]: Done in 1e47713.

File: _sass/_components/blog-post.scss (hunk 7 above)
Comment (maya) [inline _sass/_components/blog-post.scss:1]: Done in 85af45a 9ae9a19 ea7e14b fa9b3ef.

File: _includes/related-posts.html (hunk 1 above)
Comment (maya) [inline _includes/related-posts.html]: Done in 84c13c2.

File: _includes/related-posts.html (hunk 2 above)
Comment (maya) [inline _includes/related-posts.html]: I have no idea! I found this code online here: http://zhangwenli.com/blog/2014/07/15/jekyll-related-posts-without-plugin/ and it worked.

File: _sass/_components/layout.scss (hunk 10 above)
Comment (msecret) [inline _sass/_components/layout.scss:64]: Would this change once we change the imports so we can override the USWDS variables?

Comment (maya) [inline _sass/_components/layout.scss:64]: No, this is a separate thing from that issue - this cascades light link and button styles when you're using a dark background.

File: _layouts/post.html (hunk 3 above)
Comment (elainekamlley) [inline _layouts/post.html]: Hmmm, @gboone any thoughts ?

File: _layouts/tag_page.html (hunk 5 above)
Comment (elainekamlley) [inline _layouts/tag_page.html]: Ohh, yea. I like that naming better.

File: _sass/_components/blog-index.scss (hunk 6 above)
Comment (elainekamlley) [inline _sass/_components/blog-index.scss:1]: 👍

File: _sass/_components/blog-post.scss (hunk 7 above)
Comment (elainekamlley) [inline _sass/_components/blog-post.scss:1]: Woo!

File: _sass/_components/blog-post.scss (hunk 8 above)
Comment (elainekamlley) [inline _sass/_components/blog-post.scss:37]: Ah, okay. I see that now.

File: _sass/_components/layout.scss (hunk 10 above)
Comment (elainekamlley) [inline _sass/_components/layout.scss:64]: Ah, okay. I'm good with this then. @msecret any other thoughts?

File: _layouts/tag_page.html (hunk 5 above)
Comment (maya) [inline _layouts/tag_page.html]: Done in 09a6397.

File: _layouts/post.html (hunk 3 above)
Comment (gboone) [inline _layouts/post.html]: I feel like the `page` variable should be available for a single blog post. The `post` variable would be available in an archive view. I might have missed some context but if you want to use the same include in both situations, I think you can sniff that out with the `include.context` variable. Something like: `{% if include.context == "page" && page.authors %}` might work. Let me see if I can find some docs on that.

Comment (gboone) [inline _layouts/post.html]: Not docs, but here's an example of us doing it on 18f.gsa.gov. https://github.com/18F/18f.gsa.gov/blob/staging/_includes/byline.html#L1

Comment (maya) [inline _layouts/post.html]: @gboone if you know how to do this, would you mind making the change on this branch?

Comment (maya) [inline _layouts/post.html]: Fixed in 375c80a. Thanks @gboone! 

cc: @elainekamlley

**Comment by elainekamlley** [comment]
Awesome work @maya and @gboone! Merging!

**Pull request merged**